src/bindings/python/flux/job/list.py

```diff
@@ -140,7 +141,8 @@ def get_jobs(self):
                 jobs.append(child.get_job())
             except EnvironmentError as err:
                 if err.errno == errno.ENOENT:
-                    msg = f"JobID {child.jobid.orig} unknown"
+                    id = JobID(child.jobid)
+                    msg = f"JobID {id.orig} unknown"
```

Comment:
I wonder if elsewhere in the code might assume ids are instances of `JobID`. Maybe the `JobList` constructor should instead attempt to convert all `ids` in the input array to `JobID`, e.g.

```diff
diff --git a/src/bindings/python/flux/job/list.py b/src/bindings/python/flux/job/list.py
index b0ff9710f..c78bd8f2f 100644
--- a/src/bindings/python/flux/job/list.py
+++ b/src/bindings/python/flux/job/list.py
@@ -211,7 +211,7 @@ class JobList:
         self.since = since
         self.name = name
         self.queue = queue
-        self.ids = ids
+        self.ids = list(map(flux.job.JobID, ids))
         self.errors = []
         for fname in filters:
             for x in fname.split(","):
```

Reply:
I considered this approach as well, but I figured since the only failing part was the `id.orig` I wanted to avoid the unnecessary iteration.  But you are right, it would prevent other issues that we aren't expecting and future issues as well.